### ISSUE TITLE
[Spike] StripeSCA Subs - Payments taken twice for some banks

### ISSUE THREAD

**Issue by lin-d-hop** [openfoodfoundation/openfoodnetwork#7495]
Proposing a timeboxed spike to investigate this.... 

## Description

Stripe users with specific banks (Monzo, Lloyds) are reporting that Stripe payments for their subscriptions are taken twice.
The second payment is refunded within a month, but we've had a number of users payments fail as the second payment takes them into negative balance or other complications. A lot of subs users are getting very stressed by this.

## Expected Behavior

Subs just take one payment - the payment hold and the actual payment are not separate transactions taken twice.

## Actual Behaviour

Some banks are taking two payments.

## Steps to Reproduce

1.
2.
3.
4.

## Workaround

- Tell the user to wait a month to have their money returned (with weekly orders this is crazy)
- Manually refund the second payment

## Severity

S2

## Your Environment

* Version used: Related to Stripes SCA implementation
* Browser name and version:
* Operating System and version (desktop or mobile):

## Possible Fix

I don't know if this is fixable at all. It might be a bank issue or a Stripe issue. Hence, can we timebox a spike to 2 hrs to see what we can learn about our responsibility here and what a potential fix would be, if one exists.

## More Info

More sites seem to suggest using idempotent requests.... are we?
https://stripe.com/docs/api/idempotent_requests

Some interesting discussions on this thread - this one seems potentially relevant:
https://forums.wpsharks.com/t/stripe-causing-double-payments/7002/41

**Comment by lin-d-hop** [comment]
Specific order details that have been effected are to come.... @lbwright22

**Comment by lbwright22** [comment]
R455286758: 5th April, **Subs generated payment**, 27th April shows under customer's statement as 2 payments, Stripe shows one succeeded and one cancelled
R438076302: 12 April, **payment taken in Stripe** (not subs), 27th April shows on customer's statement as one refunded and one successful payment, Stripe show just one payment ('Succeeded')
R41004715: 19th April -  **Subs generated payment**, 27th April shows under customer's statement as 2 payments, Stripe shows one succeeded and one cancelled
R278418570: 26th April  **Subs generated payment**, Stripe shows one 'failed' and one 'uncaptured' payment, customer reports one payment taken and one failed.
screen shot from Stripe account

![monzostripe1](https://user-images.githubusercontent.com/45104035/116231363-aa38bf00-a750-11eb-97fb-953e18912a7b.jpg)

Customer's card account screen shot 5th April
![image](https://user-images.githubusercontent.com/45104035/116231504-e10ed500-a750-11eb-91e4-ee9aa89c0151.png)

**Comment by lbwright22** [comment]
Note: the work around in the 2nd bullet point above is not possible. The enterprise only gets the money for one transaction. Hence they can't manually refund the customer

**Comment by andrewpbrett** [comment]
I can look into this. Thanks for those details @lbwright22. It may end up having the same root cause as https://github.com/openfoodfoundation/openfoodnetwork/issues/7417, we will see.

**Comment by andrewpbrett** [comment]
One preliminary finding: I looked through the logs on Stripe for order #R000783123. We make one call to authorize the payment, and one to make the purchase. In terms of ActiveMerchant, the only difference between those two call is that the `capture_method` param is set to 'manual' in the first call and 'automatic' in the second. 

According to https://stripe.com/docs/payments/capture-later#cancel-authorization, this is the correct way to indicate that you only want to authorize the card. So the fact that there's two actual charges to the card showing up on the customer's statement seems buggy (or at least unexpected) on Stripe's part.

The first request (to authorize) expires after 7 days. I think there's a way that we could consolidate and only make one request; the reason they're split up at all is to allow for the flow where additional SCA auth is required of the customer. When there's no auth required, we proceed immediately to making the second request, but I'm thinking that we could instead call `capture` on the PaymentIntent.
